proposals/interface-types/working-notes/choices/webidl.md

```diff
@@ -0,0 +1,257 @@
+contains two integers also (for example, the age of the person and an index into
+a street address table).
+
+A given pair of integers might be either a `Chair` or a `Person` (or a 2D point)
+and there is no way of conveying the distinction unless the nominal type is also
+taken into consideration.
+
+### Algebraic Type Definitions
+
+The IT type language permits the definition of a limited form of _algebraic type
+definition_. This is, in part, to regularize concepts such as nullability and to
+permit limited forms of polymorphism (allowing a color to be named by a string,
+a vector of small integers or a large integer, for example).
+
+WebIDL explicitly prohibits the formation of new types; indeed it uses type
+unions to achieve similar polymorphism. However, as we note above, not all
+programming languages support dynamic types or type unions; which makes it
+problematic to support such polymorphism.
+
+The form of algebraic types envisaged in the IT system is intentionally
+limited. For example, we do not permit _recursive_ types to be defined. This is
+in keeping with the vision of supporting APIs across ownership boundaries where
+complex data is likely to take limited form.[^This design choice may be
+revisited in future versions of the IT specification if supporting structured
+types such as abstract XML and/or JSON become important.]
+
+### Working with WebIDL
+
+While the IT language introduces its own IDL, it is still clearly important to
+support WebIDL; particularly for Web APIs.
+
+The strategy that we adopt can be summarized as interpreting WebIDL as an
+alternate target to webAssembly itself.
+
+When using IT to specify a public API for a webAssembly module, the author must
+provide special _adapters_ -- import adapters for accessing external APIs and
+export adapters when offering a public API.
+
+Similarly, in the case where a WebIDL API is involved, there would be adapters
+between the IT language and WebIDL. There would also be adapters between WebIDL
+and the host environment; although these are usually private to the host.
+
+In many cases, the adapters would be automatically generated by language
+specific tools; which would mean that individual code authors would likely be
+not involved in the construction of the adapters.
+
+The IT language has been designed to be a slight extension of a simplification
+of WebIDL. We anticipate that it should be straightfoward to construct import
+and export adapters between IT and WebIDL completely automatically.
+
+Note that this does not imply that the semantics of IT and WebIDL are completely
+aligned: any automatically generated adapters would likely ignore those aspects
+of WebIDL that are not accounted for in the IT language.
+
+#### Modifying WebIDL itself
+
+One of the strategies that we considered was modifying WebIDL itself. However,
+for some of the same reasons outlined here, this is considered to be infeasible:
```

Comment:
@lukewagner had previously been talking about the possibility that a finished version of Interface Types might subsume some "lower-level" bits of WebIDL. Is that dream completely gone?

Reply:
If by subsume you mean replace, I don't think that is practicable. On the other hand, IT's language is most definitely based on a simplification of WebIDL.
So, the net net is that if your WebIDL API does not use extended attributes, and if it does not rely on JavaScript's ability to dynamically test types of values, then moving from WebIDL to IT IDL should be pretty easy.